### ISSUE TITLE
yamllint only reports wrong-indentation on the first faulty line 

### ISSUE THREAD

**Issue by tamere-allo-peter** [adrienverge/yamllint#427]
I'm working on a simple tool designed to parse yamllint's output and automatically fix a subset of encountered problems, currently only 11 warning or error conditions are automatically fixed.

It was working just fine until I stumbled upon this problem : yamllint doesn't report all faulty lines with wrong-indentation, only the first one in a block (sort of, I've not done in-depth testing).

```
$ yamllint --version
yamllint 1.26.3
```

Consider the following YAML file

```
---
- name: some task
  xml:
    path: /opt/app/myfile.xml
    xpath: "/blah/{{item}}"
    value: "false"
  with_items:
      - user
      - admin
      - server
  tags: blah
```

and now consider the output of yamllint with default options and parsable output on this file 

```
$ yamllint -f parsable blah.yml
blah.yml:8:7: [error] wrong indentation: expected 4 but found 6 (indentation)
```

I believe yamllint should report the same error on line 9 and line 10, but it only reports it on line 8 (the _- user_ line).

Is this the intended behavior or is it a bug ?

Thanks for your time.

**Comment by adrienverge** [comment]
Hello,

Yes this is intended. You can see a lot of examples of how yamllint is supposed to behave in `tests/rules/test_indentation.py`, for example: https://github.com/adrienverge/yamllint/blob/d0392b3/tests/rules/test_indentation.py#L692-L700

A way to see this is *the first line is wrongly indented, however it's consistent that the second one is indented as the first one*.

For your tool, maybe you can parse YAML tokens (especially `BlockMappingStart` and `BlockSequenceStart`) to detect all the lines that are subsequently indented?

**Comment by tamere-allo-peter** [comment]
I understand, thanks for your clarification. 

Having an option in yamllint to report all as errors even if they are consistent with the first line would solve my problem, however I don't think I'm able to send a patch for this unfortunately, and I don't even know if this would be easy to do or not :-(

In my tool, which only imports the sys and subprocess modules, I specifically want to avoid having to parse YAML. That's why I let yamllint's parsable output do the job for me, and then all I have to do is mostly to sort warning/errors by line then by column and then insert or delete characters or lines at the right place, based on the line/column reported by yamllint, and offsetting all subsequent errors accordingly.

If an additional option in yamllint is not on the horizon, then I believe I'll modify my tool to fake yamllint's output by creating errors for all subsequent lines until their indentation differs from the first erroneous one.

If I can make it to work, and if I'm allowed to make it public (not sure yet), would you accept such tool as a contribution to yamllint or do you prefer for it to be kept a separate project ? 

Have a nice day

**Comment by adrienverge** [comment]
Thanks for the proposal! In my opinion this goes beyond the scope of yamllint and would be better as a separate project.

**Comment by tamere-allo-peter** [comment]
@adrienverge ok, thanks anyway. I've been allowed to make it public under the GPLv3+, it's available from https://github.com/opt-nc/yamlfixer in case it might help someone. It's very crude at the moment though, and certainely contains bugs... Thanksfor yamllint !

**Comment by schoubi** [comment]
hi,

I just discovered your two amazing tools, thanks for that, and i'm stuck at the same point. 

I'm trying to lint some salt files, with a lot of values as this :

```
entity:
    list:
        - value:
           - server1
           - server2
           - server3
        - force: true
```

As you said, `yamllint ` with `yamlfixer ` will "fix" the file as this, only indenting the first element of the list :

```
entity:
    list:
        - value:
            - server1
          - server2
          - server3
        - force: true
```
which in fact broke yaml. 
How can i treat this correctly ?

**Comment by andrewimeson** [comment]
@schoubi it'd be better to discuss that in the [yamlfixer](https://github.com/opt-nc/yamlfixer) project than here.

It's more opinionated (and not configurable) than Jérôme's `yamlfixer`, but [`yamlfix`](https://github.com/lyz-code/yamlfix) can properly transform your first example into valid YAML that passes `yamllint`

**Comment by timansky** [comment]
Faced with another bug/feature

```
  indentation:
    spaces: 2
    indent-sequences: true
```

```
dict:
  - item1
    - item2
    - item3
```
 This yaml will pass

**Comment by adrienverge** [comment]
Hello @zymurnerd, thanks for these clear details.

However:
- What you describe is different from the original issue. The author of the issue argued that in case of wrong indentation, all lines of the sequence should be reported. For example:
  ```yaml
  fruit:
    - apple   # report an error for this line
    - orange  # report another error for this line
    - banana  # again…
  ```

- I don't think there is any bug, because the example you provide:
  ```yaml
  fruit:
  - apple
    - orange
    - banana
  ```
  is **valid YAML, but is not a 3-item sequence**, it's just a 1-item sequence with value `"apple - orange - banana"`.

  Said differently, this snippet you provided is just another way to write:
  ```yaml
  fruit:
  - apple - orange - banana
  ```
  or:
  ```yaml
  fruit: ["apple - orange - banana"]
  ```

Let me know whether this is clear :)
